18F/identity-idp
# Enhance FormSteps to manage history using path fragment

**Why**: So that I can navigate back and forth in the new IdV v2 application, and my location is maintained accurately in the address bar path.

In the original document capture implementation, we included history management as part of the "steps" implemented there. That implementation only allowed for querystring-based hash fragments (e.g. `/document_capture#step=selfie`).

Because we want the React application to manage incoming routed URLs from Rails, and because hash fragment routing is unsightly, this pull request seeks to expand this base support to allow for path-based history management.

**Testing Instructions:**

1. Set `idv_api_enabled: "true"` in local `config/application.yml`
2. Go to: http://localhost:3000/verify/v2/
3. Observe that you see the personal key page (the current first step of the application)
4. Click "Continue"
5. Observe that you see the confirmation modal, and the address bar changes to reflect `/verify/v2/personal_key_confirm`
6. Refresh the page
7. Observe that you still see the confirmation modal
8. Press your browser's back button
9. Observe that you see the personal key page, and the address bar changes to the previous value (`/verify/v2/`)

**Video:**

https://user-images.githubusercontent.com/1779930/163627803-e4a7b523-1dbe-4d4d-8481-e77643bd07e3.mov

_Draft only because I wrapped this up rather quickly at the end of the day and haven't had a chance to give it a thorough look-over, but functionally it seems to work quite fine._
